src/h5web/providers/hsds/hsds-api.ts

```diff
@@ -109,14 +109,12 @@ export class HsdsApi extends ProviderApi {
 
     // https://github.com/HDFGroup/hsds/issues/88
     // HSDS does not reduce the number of dimensions when selecting indices
-    // This is an issue when trying to slice nD arrays
-    // Therefore the dimension reduction is done "manually" here with a flat
-    const numberIndices = selection.match(/\d+/gu)?.length;
-    if (!numberIndices) {
-      return value;
+    // Therefore the flattening must be done on all dimensions regardless of the selection
+    if (hasArrayShape(entity)) {
+      return (value as unknown[]).flat(entity.shape.length - 1);
     }
```

Comment:
https://github.com/HDFGroup/hsds/issues/88 works in our favour here as `value` will always have the same number of dimensions of the original dataset, even when slicing